plugin/path_config.go

```diff
@@ -169,6 +178,44 @@ func (b *GcpAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 		return nil, fmt.Errorf("failed to persist configuration to storage: %w", err)
 	}
 
+	// Disable Automated Rotation and Deregister credentials if required
+	if c.DisableAutomatedRotation {
+		// Ensure de-registering only occurs on updates and if
+		// a credential has actually been registered (rotation_period or rotation_schedule is set)
+		deregisterReq := &rotation.RotationJobDeregisterRequest{
+			MountType: req.MountType,
+			ReqPath:   req.Path,
+		}
+		b.Logger().Debug("Deregistering rotation job", "mount", req.MountPoint+req.Path)
+		err := b.System().DeregisterRotationJob(ctx, deregisterReq)
+		if err != nil {
+			return logical.ErrorResponse("error de-registering rotation job: %s", err), nil
+		}
+	} else {
+		// Now that the root config is set up, register the rotation job if it's required.
+		if c.ShouldRegisterRotationJob() {
+			cfgReq := &rotation.RotationJobConfigureRequest{
+				Name:             rootRotationJobName,
```

Comment:
Could Vault itself set this field for the plugin developer? Is there any reason we want the plugin developer to provide a name?

Reply:
Good point. We were previously using this field more in the DB Engine, where the unique DB Connection Name was used by the RM to let the DB Backend know which DB Connection to rotate. However, I did realize that that behavior was not required and we could parse the RotationID sent by the RM to gather the requisite info. With that in mind, the `Name` parameter may be unused. Can confirm this and we should be able to remove this from the plugins

Reply:
Confirming that `Name` is unused in Automated Root Rotation. We have kept it around in Vault Core because it may come in handy in the future when expanding support for static credentials and other types of secrets.